### PR TITLE
Move visual-diff-plugin to testing repo

### PR DESCRIPTION
**Pull request by bearfriend**
Copied from https://github.com/BrightspaceUI/core/pull/3610

### CODE REVIEW

**Review by dlockhart — APPROVED** [review]
This is probably sufficient for now and we can add tests once this is actually doing something useful.

File: src/server/visual-diff-plugin.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				return;
+			}
+			if (session.browser.type !== 'playwright') {
+				throw new Error('Visual-diff is only supported for browser type Playwright');
+			}
+
+			const browser = session.browser.name.toLowerCase();
+			const { dir, newName } = extractTestPartsFromName(payload.name);
+			const goldenFileName = `${join(dirname(session.testFile), 'screenshots', ciDir, 'golden', browser, dir, newName)}.png`;
+			const currentFileName = `${join(dirname(session.testFile), 'screenshots', ciDir, 'current', browser, dir, newName)}.png`;
```
Comment (svanherk) [inline src/server/visual-diff-plugin.js:52]: Doesn't need to be figured out now, but based on the crazy names we've seen in the past hitting Windows limits, wondering if we want to make this shorter.  I'm not particularly attached to `screenshots`

**Review by svanherk — APPROVED** [review]

File: src/server/visual-diff-plugin.js (hunk 1 above)
Comment (dlockhart) [reply src/server/visual-diff-plugin.js:52]: That's a good point -- could be `vdiff`?

Comment (dlockhart) [reply src/server/visual-diff-plugin.js:52]: I'm good with being consistent everywhere with `vdiff`.

Comment (bearfriend) [reply src/server/visual-diff-plugin.js:52]: Merging as-is for now, and will address this probably in the golden migration PR.

**Pull request merged**